src/models/blockchain/transaction/primitive_witness.rs

```diff
@@ -1546,11 +1671,121 @@ mod test {
         }
     }
 
+    #[proptest(cases = 4, async = "tokio")]
+    async fn updating_primitive_witness_with_ms_data_works(
+        // Notice only SingleProof-backed txs need inputs to allow updating, not PW-backed ones.
+        #[strategy(0usize..20)] _num_inputs_own: usize,
+        #[strategy(0usize..20)] _num_outputs_own: usize,
+        #[strategy(0usize..20)] _num_public_announcements_own: usize,
+        #[strategy(0usize..20)] _num_inputs_mined: usize,
+        #[strategy(0usize..20)] _num_outputs_mined: usize,
+        #[strategy(0usize..20)] _num_public_announcements_mined: usize,
+        #[strategy(PrimitiveWitness::arbitrary_tuple_with_matching_mutator_sets(
+            [(#_num_inputs_own, #_num_outputs_own, #_num_public_announcements_own),
+            (#_num_inputs_mined, #_num_outputs_mined, #_num_public_announcements_mined),],
+    ))]
+        pws: [PrimitiveWitness; 2],
+    ) {
+        let [own_pw, mined_pw] = pws;
+
+        let kernel_hash = own_pw.kernel.mast_hash();
+        prop_assert!(
+            TransactionProof::Witness(own_pw.clone())
+                .verify(kernel_hash)
+                .await
+        );
+
+        let ms_update = MutatorSetUpdate::new(
+            mined_pw.kernel.inputs.clone(),
+            mined_pw.kernel.outputs.clone(),
+        );
+        let updated_pw = PrimitiveWitness::update_with_new_ms_data(own_pw, ms_update);
+
+        let new_kernel_hash = updated_pw.kernel.mast_hash();
+        prop_assert!(
+            TransactionProof::Witness(updated_pw)
+                .verify(new_kernel_hash)
+                .await
+        );
+    }
+
+    #[traced_test]
+    #[tokio::test]
+    async fn arb_is_valid_unit_test_small() {
```

Comment:
`arb` in the context of proptest and test-strategy means something different from the thing you're testing. Why not spell it out?